### PR TITLE
add deletion orphan for mch manifests and fix acm detection

### PR DESCRIPTION
**Pull request by zhiweiyin318**
Signed-off-by: Zhiwei Yin <zyin@redhat.com>
tasks:  
https://github.com/stolostron/backlog/issues/26299
https://github.com/stolostron/backlog/issues/26467

### CODE REVIEW

File: agent/pkg/controllers/clusterclaim_controller.go [hunk 1]
```diff
@@ -27,73 +29,112 @@ type clusterClaimController struct {
...
+	if mch == nil {
+		return ctrl.Result{}, updateClusterClaim(ctx, c.client,
+			constants.HubClusterClaimName, constants.HubNotInstalled)
 	}
-	if err != nil {
+
+	hubValue := constants.HubInstalledWithoutSelfManagement
+	if mch.GetLabels()[constants.GlobalHubOwnerLabelKey] == constants.GlobalHubOwnerLabelVal {
+		hubValue = constants.HubInstalledByHoH
+	} else if !mch.Spec.DisableHubSelfManagement {
```
Comment (clyang82) [inline agent/pkg/controllers/clusterclaim_controller.go:51]: Maybe update judgement
if mch.GetLabels()[constants.GlobalHubOwnerLabelKey] == constants.GlobalHubOwnerLabelVal {
if !mch.Spec.DisableHubSelfManagement {
hubValue = constants.HubInstalledWithSelfManagement
} else {
hubValue = constants.HubInstalledByHoH
}
}

File: agent/pkg/controllers/clusterclaim_controller.go [hunk 2]
```diff
@@ -27,73 +29,112 @@ type clusterClaimController struct {
...
 	}
-	reqLogger.Info("Reconciliation complete.")
-	return ctrl.Result{}, nil
+
+	// requeue to wait the acm version is available
+	return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 5}, nil
 }
 
-func getACMVersion(ctx context.Context, client client.Client, namespacedname types.NamespacedName) (string, error) {
+func getMch(ctx context.Context, client client.Client, namespacedname types.NamespacedName) (*mchv1.MultiClusterHub, error) {
```
Comment (clyang82) [inline agent/pkg/controllers/clusterclaim_controller.go:67]: for abbreviation, just capital all make more sense.

File: operator/pkg/controllers/addon/addoninstall_controller.go [hunk 3]
```diff
@@ -82,6 +82,12 @@ func (r *HoHAddonInstallReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		},
 	}
 
+	// install addon in open-cluster-management-global-hub-system ns if the cluster has local-cluster.
+	switch getHub(cluster) {
+	case commonconstants.HubInstalledWithSelfManagement:
+		addon.Spec.InstallNamespace = "open-cluster-management-global-hub-system"
```
Comment (morvencao) [inline operator/pkg/controllers/addon/addoninstall_controller.go:88]: replace `"open-cluster-management-global-hub-system"` with `commonconstants.HOHSystemNamespace` ?

File: pkg/constants/constants.go [hunk 4]
```diff
@@ -105,4 +105,13 @@ const (
...
 	VersionClusterClaimName = "version.open-cluster-management.io"
+
+	// HubClusterClaimName is a claim to record the ACM Hub
+	HubClusterClaimName = "hub.open-cluster-management.io"
+
+	// the value of the HubClusterClaimName ClusterClaim
+	HubNotInstalled                   = "NotInstalled"
+	HubInstalledWithSelfManagement    = "InstalledWithSelfManagement"
+	HubInstalledWithoutSelfManagement = "InstalledWithoutSelfManagement"
+	HubInstalledByHoH                 = "InstalledByHoH"
```
[the remark this answers is not in the document]
Comment (zhiweiyin318) [reply pkg/constants/constants.go:116]: done

File: agent/pkg/controllers/clusterclaim_controller.go [hunk 5]
```diff
@@ -27,73 +29,112 @@ type clusterClaimController struct {
...
-	if request.Namespace != "" {
-		var err error
-		version, err = getACMVersion(ctx, c.client, request.NamespacedName)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
+	var err error
+	mch, err := getMch(ctx, c.client, request.NamespacedName)
+	if err != nil {
+		reqLogger.Error(err, "failed to get ACM version")
```
[the remark this answers is not in the document]
Comment (zhiweiyin318) [reply agent/pkg/controllers/clusterclaim_controller.go]: done

File: agent/pkg/controllers/clusterclaim_controller.go (hunk 2 above)
Comment (zhiweiyin318) [reply agent/pkg/controllers/clusterclaim_controller.go]: done

File: operator/pkg/controllers/addon/addoninstall_controller.go (hunk 3 above)
Comment (zhiweiyin318) [reply operator/pkg/controllers/addon/addoninstall_controller.go]: done

File: agent/pkg/controllers/clusterclaim_controller.go (hunk 1 above)
Comment (zhiweiyin318) [reply agent/pkg/controllers/clusterclaim_controller.go:51]: the logic like this: 
if the ACM is installed by HoH, set value = HubInstalledByHoH
if the ACM is not installed by HoH, set value = HubInstalledWithSelfManagement or HubInstalledWithoutSelfManagement.

Comment (clyang82) [reply agent/pkg/controllers/clusterclaim_controller.go:51]: Yes. sound good for me.

**Pull request merged**